### PR TITLE
Create virtual_dev_environments.md

### PR DESCRIPTION
**Pull request by anenadic**
A new task to explain the use of virtual development environments.

It uses Python often for examples, so not sure that is good or what we want - perhaps this is meant to be a language agnostic resource and we can link to other programming language specific task pages. 

I added expernal links for R and Python too and we can add for other languages used by science clusters (which are those?) so I can remove the mentioning of Python and Python specific examples from the text - what do you think @egpbos @dgarijo ?

### CODE REVIEW

File: pages/your_tasks/virtual_dev_environments.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+related_pages: []
+---
+
+## What are virtual software development environments?
+
+A virtual software development environment helps us create an **isolated working copy** of a software project that uses a specific 
+version of a programming language interpreter/compiler (e.g. Python 3.11) together with specific versions of a number of external 
+libraries (dependencies) required by our software installed into that virtual environment. 
+
+Virtual environments are typically implemented as directories with a particular structure (usually contained within software projects
```
Comment (shoaibsufi) [inline pages/your_tasks/virtual_dev_environments.md:15]: I am not sure the text (usually contained within software projects ...) adds to the explanation - consider removing or simplifying?

File: pages/your_tasks/virtual_dev_environments.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+The solution for this problem is to create a self-contained virtual environment per project, which contains a particular version of your 
+programming language interpreter/compiler plus a number of additional external libraries.
+
+Another big motivator for using virtual environments is that they make **sharing your code with others** (users or developers) much easier.
+By sharing a description of your virtual development environment you enable others to quickly replicate the same environment 
+on their machines and run or further develop your software - making your work **portable**, **reusable** and more **reproducible**.
+
+### Considerations
+
+- As more external libraries are added to your software project over time, you can add them to its specific virtual environment
```
Comment (shoaibsufi) [inline pages/your_tasks/virtual_dev_environments.md:39]: Is it worth linking out to this as a bad example of a spaghetti setup when things are not isolated - https://xkcd.com/1987/

File: pages/your_tasks/virtual_dev_environments.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+- Make your research software reusable and your research that relies on that software reproducible by setting up and sharing its virtual development environment.
+
+## How do you create virtual software development environments? 
+
+### Description
+
+Most modern programming languages use some kind of virtual environments or a similar mechanism to isolate libraries or dependencies for a specific project, 
+making it easier to develop, run, test and share code with others. 
+Some examples include Bundler for Ruby, Conan for C++, or Maven with classpath for Java.
```
Comment (shoaibsufi) [inline pages/your_tasks/virtual_dev_environments.md:65]: We mentioned python a lot - so might be worth mentioning the one for python here (eps. given it's popularity)

File: pages/your_tasks/virtual_dev_environments.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+In addition to a package manager, you will need another command line tool to create and manage virtual environments on your machine. 
+Sometimes, a package manager combines both of these functionalities and you only need to install one extra tool on your system.
+
+### Considerations
+
+- There are often multiple package and environment management tools for a single programming language. For example,
+there are several commonly used command line tools for managing Python packages and virtual environments to choose from:
+  - `pip` - Python package manager tool which interacts and obtains the packages from the central repository called Python Package Index (PyPI). `pip` can now be used with all Python distributions (including Anaconda).
+  - `venv` - Python virtual environment manager tool available by default from the standard Python distribution from Python 3.3+
+  - `venv` - Python virtual environment manager tool available by default from the standard Python distribution from Python 3.3+
```
Comment (shoaibsufi) [inline pages/your_tasks/virtual_dev_environments.md:80]: Duplicate line?

File: pages/your_tasks/virtual_dev_environments.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+  - `virtualenv` - Python virtual environment manager tool which needs to be installed separately (does not come with Python distribution) but supports both Python 2.7+ and Python 3.3+ versions
+  - `pipenv` - Python package and environment management sysyem created to fix certain shortcomings of `virtualenv`
```
Comment (shoaibsufi) [inline pages/your_tasks/virtual_dev_environments.md:82]: might be worth adding a note that pipenv and pip-venv are not the same thing (and maybe a line on how they are different)

**Review by shoaibsufi — APPROVED** [review]

File: pages/your_tasks/virtual_dev_environments.md (hunk 4 above)
Comment (anenadic) [inline pages/your_tasks/virtual_dev_environments.md:80]:
```suggestion
  - `venv` - Python virtual environment manager tool available by default from the standard Python distribution from Python 3.3+
```

File: pages/your_tasks/virtual_dev_environments.md (hunk 5 above)
Comment (anenadic) [inline pages/your_tasks/virtual_dev_environments.md:82]:
```suggestion
  - `pipenv` (different from both `pip` and `venv`) - Python package and environment management sysyem created to fix certain shortcomings of `virtualenv`
```

File: pages/your_tasks/virtual_dev_environments.md (hunk 2 above)
Comment (anenadic) [reply pages/your_tasks/virtual_dev_environments.md:39]: Fixed via [4457d22](https://github.com/EVERSE-ResearchSoftware/RSQKit/commit/4457d22ad92f2fccc4274076b0e01d5a8f192f12).

File: pages/your_tasks/virtual_dev_environments.md (hunk 1 above)
Comment (anenadic) [reply pages/your_tasks/virtual_dev_environments.md]: Clarified via 4f3c01d

**Comment by egpbos** [comment]
Sorry for the lack of response @anenadic! I somehow overlooked the personal pings (need to adjust my notifications so things don't get lost...).

**Comment by anenadic** [comment]
No problem @anenadic - here is another PR to address the above: https://github.com/EVERSE-ResearchSoftware/RSQKit/pull/40

**Pull request merged**